homeassistant/components/climate/__init__.py

```diff
@@ -562,16 +562,15 @@ def max_humidity(self):
 
     def _convert_for_display(self, temp):
         """Convert temperature into preferred units for display purposes."""
-        if temp is None or not isinstance(temp, Number):
+        if (temp is None or not isinstance(temp, Number) or
+                self.temperature_unit == self.unit_of_measurement):
             return temp
 
         value = convert_temperature(temp, self.temperature_unit,
                                     self.unit_of_measurement)
 
         if self.unit_of_measurement is TEMP_CELSIUS:
-            decimal_count = 1
+            return round(value, 1)
         else:
             # Users of fahrenheit generally expect integer units.
-            decimal_count = 0
-
-        return round(value, decimal_count)
+            return int(value)
```

Comment:
It was already doing so by setting decimal_count to 0. However, there is a subtle difference in Python:

```python
>>> round(12.4, 0)
12.0
>>> round(12.4)
12
```